lib/pf/radius/rest.pm

```diff
@@ -0,0 +1,89 @@
+package pf::radius::rest;
+
+=head1 NAME
+
+pf::radius::rest
+
+=cut
+
+=head1 DESCRIPTION
+
+Helper methods to use REST with FreeRADIUS
+
+=cut
+
+use strict;
+use warnings;
+
+use pf::log;
+use Apache2::Const -compile =>
+  qw(DONE OK DECLINED HTTP_UNAUTHORIZED HTTP_NOT_IMPLEMENTED HTTP_UNSUPPORTED_MEDIA_TYPE HTTP_PRECONDITION_FAILED HTTP_NO_CONTENT HTTP_NOT_FOUND SERVER_ERROR HTTP_OK HTTP_INTERNAL_SERVER_ERROR);
+use pf::api::error;
+
+=head2 format_response
+
+Format a PacketFence RADIUS response to the format expected by the FreeRADIUS REST module
+
+=cut
+
+sub format_response {
+    my ($response) = @_;
+
+    my $radius_return = shift @$response; 
+    my %mapped_object = @$response; 
+    %mapped_object = ( %{$mapped_object{"RADIUS_AUDIT"}}, %mapped_object);
+
+    delete $mapped_object{"RADIUS_AUDIT"};
+    get_logger->trace(sub { use Data::Dumper ; "RADIUS REST object : ". Dumper(\%mapped_object) });
+    $response = \%mapped_object; 
+    
+    unless ($radius_return == 2) { 
+        die pf::api::error->new(status => Apache2::Const::HTTP_UNAUTHORIZED, response => $response);
+    }
```

Comment:
Right but we will need to map the return code from Switch.pm to an HTTP code which I'm not sure about